cdap-ui/app/features/tracker/controllers/audit-ctrl.js

```diff
@@ -83,18 +88,111 @@ class TrackerAuditController {
     this.myTrackerApi.getAuditLogs(params)
       .$promise
       .then((response) => {
+        this.isTrackerSetup = true;
         this.auditLogs = response;
       }, (err) => {
-        console.log('Error: ', err);
-        this.myAlertOnValium.show({
-          type: 'danger',
-          content: 'Enable the Tracker extension to view audit logs.'
-        });
+        if (err.statusCode === 503) {
+          this.isTrackerSetup = false;
+        } else {
+          this.myAlertOnValium.show({
+            type: 'danger',
+            content: err.data
+          });
+        }
       });
   }
+
+  enableTracker() {
+    this.enableTrackerLoading = true;
+    this.myTrackerApi.getTrackerApp({
+      namespace: this.$state.params.namespace,
+      scope: this.$scope
+    })
+    .$promise
+    .then(() => {
+      // start programs
+      this.startPrograms();
+    }, () => {
+      // create app
+      this.createTrackerApp();
+    });
+  }
+
+  createTrackerApp() {
+    this.myTrackerApi.getCDAPConfig({ scope: this.$scope })
+      .$promise
+      .then((res) => {
+        let zookeeper = res.filter( (c) => {
+          return c.name === 'zookeeper.quorum';
+        })[0].value;
+
+        let kafkaNamespace = res.filter( (c) => {
+          return c.name === 'kafka.zookeeper.namespace';
+        })[0].value;
+
+        let numPartitions = res.filter( (c) => {
+          return c.name === 'kafka.num.partitions';
+        })[0].value;
+
+        let topic = res.filter( (c) => {
+          return c.name === 'audit.kafka.topic';
+        })[0].value;
+
+        let config = {
+          artifact: {
+            name: 'tracker',
+            version: '1.0-SNAPSHOT',
```

Comment:
Ideally no, but this is the same case for hydrator plugins version, and cloudera navigator. The ultimate correct behavior should be having a UI Config that will define all these parameters